pkg/sampling/encoding_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sampling
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func must[T any](t T, err error) T {
+	if err != nil {
```

Comment:
could you use `require.NoError()` here instead of panicking?

Reply:
As a matter of syntax, this technique allows one-line tests below. I think it's an overall readability improvement.  Otherwise, I have to replace each `must(functionCall())` with
```
  _, err := functionCall()
  require.NoError(t, err)
```

that feels like a net loss to me, what do you think?